lib/Auth/Source/privacyidea.php

```diff
@@ -234,6 +236,8 @@ protected function login_chal_resp($username, $password, $transaction_id, $signa
         /* If we get this far, we have a valid login. */
         $attributes = array();
         $arr = array("username", "surname", "email", "givenname", "mobile", "phone", "realm", "resolver");
+        // Add all additional attributes defined in the array map to the search array
+        $arr = array_merge(array_diff(array_keys($this->attributemap), $arr), $arr);
```

Comment:
You are completely right, I did not read the documentation thoroughly enough. Plus, the documentation needs updating to reflect the changed behaviour. I'll make a new commit tomorrow.